tests/integration-tests/tests/networking/test_security_groups.py

```diff
@@ -106,19 +122,30 @@ def test_overwrite_sg(region, scheduler, custom_security_group, pcluster_config_
         _assert_security_group_rules(
             ec2_client,
             mount_target_security_group_id,
-            [mount_target_security_group_id, custom_security_group_id],
+            [mount_target_security_group_id, custom_security_group_ids[0]],
         )
-
     if scheduler == "slurm":
         logging.info("Checking SSH connection between cluster nodes before cluster update")
         _check_connections_between_head_node_and_compute_nodes(cluster)
+
+        # Update the Security Groups of HN to check if the HN instance is replaced.
+        updated_config_file = pcluster_config_reader(
+            config_file="pcluster.config.update_sg.yaml",
+            fsx_name=fsx_name,
+            efs_name=efs_name,
+            vpc_security_group_id=custom_security_group_ids,
+            additional_security_groups=additional_security_groups,
+        )
+        cluster.update(str(updated_config_file), force_update="true")
```

Comment:
do we really need an extra update step? every update will take 15/30mins, I'd unify the changes in the `pcluster.config.update_sg.yaml` in the already existing `pcluster.config.update.yaml` file.

We can perform the update just one time and do all the checks we need.

Reply:
The reason for creating another yaml file itself, was because one of the checks in the `pcluster.config.update.yaml` is that the HeadNodes' managed sg allows access to Complte fleets custom SG

Reply:
Will make changes in the `test_additional_sg_and_ssh_from` tests for updating the cluster with a new sg_id for AdditionalSecurityGroups and SecurityGroups. However, this will not reduce the time for the tests a I would still need to update the cluster.